src/libraries/Common/src/System/Net/ArrayBuffer.cs

```diff
@@ -41,11 +41,11 @@ public void Dispose()
             _activeStart = 0;
             _availableStart = 0;
 
+            byte[] array = _bytes;
+            _bytes = null!;
+
             if (_usePool)
             {
-                byte[] array = _bytes;
-                _bytes = null!;
-
                 if (array != null)
```

Comment:
You may combine it to the if (_usePool condition since the rest is gone now.

Reply:
Done